hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector.java

```diff
@@ -43,6 +43,8 @@
     private static final long ONE_SECOND_IN_NANOS = TimeUnit.SECONDS.toNanos(1);
     private static final long SLOW_OPERATION_THREAD_MAX_WAIT_TIME_TO_FINISH = TimeUnit.SECONDS.toMillis(10);
 
+    private static boolean isFirstLog = true;
```

Comment:
Perhaps better not to make this static. Let each HZ instance tell it once. Also if it is static, you have threading issues here (no happens before relation).